subprojects/test-kit/src/integTest/groovy/org/gradle/testkit/runner/GradleRunnerGradleVersionIntegrationTest.groovy

```diff
@@ -73,9 +73,39 @@ class GradleRunnerGradleVersionIntegrationTest extends BaseGradleRunnerIntegrati
 
         where:
         version                      | configurer
-        buildContext.version.version | { if (!GradleContextualExecuter.embedded) { it.withGradleInstallation(buildContext.gradleHomeDir) }}
-        VERSION                      | { it.withGradleDistribution(locator.getDistributionFor(GradleVersion.version(VERSION))) }
-        VERSION                      | { it.withGradleVersion(VERSION) }
+        buildContext.version.version | new EmbeddedDistributionConfigurer()
```

Comment:
Another solution would be to add the name manually, right? So having something like this as method name:
```
def "execute build with different distribution types (#description)"() {
```

Would this work?

Reply:
`def "execute build with different distribution types #type"() {` seems to work, though.